presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -96,12 +96,12 @@ public double overlapPercentWith(StatisticRange other)
     {
         requireNonNull(other, "other is null");
 
-        if (this.equals(other)) {
-            return 1.0;
+        if (this.isEmpty() || other.isEmpty() || this.distinctValues == 0 || other.distinctValues == 0) {
+            return 0.0; // zero is better than NaN as it will behave properly for calculating row count
```

Comment:
If both are empty, they overlap 100%. Please explain

Reply:
From the practical perspective it doesn't matter what you return if both ranges are empty. As if both of them are empty, it doesn't matter what the filter factor you return. The result will be empty. So, we could return either `0` or `1`.

When one of the ranges is not empty and the other is empty - the filter factor should be 0. Because you don't know what cardinallity will be multiplied by this intersection factor. That's the way it was actually.

But for me it seems to be weird, that `empty X non empty -> 0`, `non empty X empty -> 0` but `empty X empty` -> 1. 

So, for consistency i decided to change that. 

But in practice it doesn't change anything. I agree. Neither of existing test were affected by this change.